wgpu-hal/src/gles/mod.rs

```diff
@@ -295,6 +295,13 @@ pub struct Device {
     counters: wgt::HalCounters,
 }
 
+impl Drop for Device {
+    fn drop(&mut self) {
+        let gl = &self.shared.context.lock();
+        unsafe { gl.delete_vertex_array(self.main_vao) };
```

Comment:
question: Why was the body of `gles::Device` not migrated in its entirety here?

answer: Oh, I see that it got split between `impl Drop for Device` and `impl Drop for Queue`.

question: This means that these operations were re-ordered a bit. Is this okay?

Reply:
I checked and it's fine.